core-bundle/src/Search/Document.php

```diff
@@ -90,6 +90,30 @@ public function getContentCrawler(): Crawler
         return $this->crawler;
     }
 
+    public function extractCanonicalUri(): ?UriInterface
+    {
+        foreach ($this->getHeaders() as $key => $values) {
+            if ('link' === $key) {
+                foreach ($values as $value) {
+                    if (preg_match('@<(https?://(.+))>; rel="canonical"@', $value, $matches)) {
```

Comment:
```suggestion
                    if (preg_match('@<(https?://(.+))>;\s*rel="canonical"@', $value, $matches)) {
```

as far as I can see from the spec, the space before `rel=` seems to be optional? Not sure if spaces between around `=` are allowed too? `rel = "canonical"`